### PR TITLE
Bump astarte device sdk v0.8

### PR DESCRIPTION
**Pull request by rgallor**

### CODE REVIEW

File: src/data/mod.rs [hunk 1]
```diff
@@ -119,17 +160,15 @@ pub mod tests {
...
 
-    mock! {
-        pub Subscriber {}
         #[async_trait]
-        impl Subscriber for Subscriber {
-             async fn on_event(&mut self) -> Option<Result<AstarteDeviceDataEvent, AstarteError>>;
-             async fn exit(self) -> Result<(), AstarteError>;
+        impl Subscriber for PubSub {
+             async fn recv(&self) -> Result<DeviceEvent, AstarteError>;
+
```
Comment (harlem88) [inline src/data/mod.rs:167]:
```suggestion
```

**Review by harlem88 — CHANGES_REQUESTED** [review]

File: src/data/astarte_device_sdk_lib.rs [hunk 2]
```diff
@@ -97,39 +92,33 @@ impl AstarteDeviceSdkConfigOptions {
...
+            return registry
+                .read()
+                .await
+                .map_err(DeviceSdkError::ReadSecret)
+                .map(Credential::secret);
         }
 
         if let Some(token) = &self.pairing_token {
-            return self.register_device(device_id, token, registry).await;
+            // TODO: when will the device be registered?
```
Comment (harlem88) [inline src/data/astarte_device_sdk_lib.rs:103]: Why do you add a TODO here? Is it something to handle in another PR?

Comment (rgallor) [reply src/data/astarte_device_sdk_lib.rs:103]: I forgot it there while rebasing! Just removed

**Review by harlem88 — APPROVED** [review]

**Comment by harlem88** [comment]
Add Astarte bump to the `changelog`.

**Review by harlem88 — APPROVED** [review]

File: Cargo.toml [hunk 3]
```diff
@@ -31,6 +31,7 @@ rust-version = "1.72.0"
 
 [dependencies]
 astarte-device-sdk = { workspace = true, features = ["derive"] }
+astarte-message-hub-proto = { workspace = true }
```
Comment (joshuachp) [inline Cargo.toml:34]:
```suggestion
```

File: src/data/astarte_message_hub_node.rs [hunk 4]
```diff
@@ -67,13 +60,20 @@ impl AstarteMessageHubOptions {
...
+            JoinHandle<Result<(), AstarteError>>,
+        ),
+        MessageHubError,
+    >
     where
         P: AsRef<Path>,
     {
-        let grpc_cfg = GrpcConfig::new(DEVICE_RUNTIME_NODE_UUID, self.endpoint.clone());
+        let endpoint = Endpoint::from_shared(self.endpoint.clone())?;
+        let grpc_cfg = GrpcConfig::new(DEVICE_RUNTIME_NODE_UUID, endpoint);
```
Comment (joshuachp) [inline src/data/astarte_message_hub_node.rs:74]:
```suggestion
        let grpc_cfg = GrpcConfig::from_url(DEVICE_RUNTIME_NODE_UUID, &self.endpoint);
```

File: src/data/astarte_message_hub_node.rs [hunk 5]
```diff
@@ -50,9 +41,11 @@ pub enum MessageHubError {
     /// missing configuration for the Astarte Message Hub
     MissingConfig,
     /// couldn't add interfaces directory
-    Interfaces(#[source] astarte_device_sdk::builder::BuilderError),
+    Interfaces(#[source] astarte_device_sdk::introspection::AddInterfaceError),
     /// couldn't connect to Astarte
     Connect(#[source] astarte_device_sdk::Error),
+    /// Invalid endpoint
+    Endpoint(#[from] astarte_message_hub_proto::tonic::transport::Error),
```
Comment (joshuachp) [inline src/data/astarte_message_hub_node.rs:48]:
```suggestion
```

**Review by joshuachp — CHANGES_REQUESTED** [review]

**Review by joshuachp — APPROVED** [review]

**Review by joshuachp — APPROVED** [review]

**Pull request merged**